ssl/ssl_key_share.cc

```diff
@@ -176,20 +178,550 @@ class X25519KeyShare : public SSLKeyShare {
   uint8_t private_key_[32];
 };
 
+class KEMKeyShare : public SSLKeyShare {
+ public:
+  KEMKeyShare(int nid, uint16_t group_id) : nid_(nid), group_id_(group_id) {}
+
+  uint16_t GroupID() const override { return group_id_; }
+
+  bool Offer(CBB *out) override {
```

Comment:
Boring changed the function names to `Encap, Decap, Generate` which makes much more sense in the KEM context (https://github.com/google/boringssl/blob/master/ssl/internal.h#L1107-L1121). Can we do something similar?

Reply:
I'm happy to make this change, but I'd prefer to do so in a separate Pull Request